assets/js/blocks/cart-checkout/payment-methods/express-payment-methods.js

```diff
@@ -48,32 +55,42 @@ const ExpressPaymentMethods = () => {
 			setPaymentStatus,
 		]
 	);
+
 	const onExpressPaymentClose = useCallback( () => {
 		setActivePaymentMethod( previousActivePaymentMethod.current );
 		if ( previousPaymentMethodData.current.isSavedToken ) {
 			setPaymentStatus().started( previousPaymentMethodData.current );
 		}
 	}, [ setActivePaymentMethod, setPaymentStatus ] );
-	const paymentMethodIds = Object.keys( paymentMethods );
-	const content =
-		paymentMethodIds.length > 0 ? (
-			paymentMethodIds.map( ( id ) => {
-				const expressPaymentMethod = isEditor
-					? paymentMethods[ id ].edit
-					: paymentMethods[ id ].content;
-				return isValidElement( expressPaymentMethod ) ? (
-					<li key={ id } id={ `express-payment-method-${ id }` }>
-						{ cloneElement( expressPaymentMethod, {
-							...paymentMethodInterface,
-							onClick: onExpressPaymentClick( id ),
-							onClose: onExpressPaymentClose,
-						} ) }
-					</li>
-				) : null;
-			} )
-		) : (
-			<li key="noneRegistered">No registered Payment Methods</li>
-		);
+
+	const content = useMemo( () => {
+		const entries = Object.entries( paymentMethods );
+
+		if ( ! entries.length ) {
+			return <li key="noneRegistered">No registered Payment Methods</li>;
+		}
+
+		return entries.map( ( [ id, paymentMethod ] ) => {
+			const expressPaymentMethod = isEditor
+				? paymentMethod.edit
+				: paymentMethod.content;
+			return isValidElement( expressPaymentMethod ) ? (
+				<li key={ id } id={ `express-payment-method-${ id }` }>
+					{ cloneElement( expressPaymentMethod, {
+						...currentPaymentMethodInterface.current,
```

Comment:
I think there may be some risk here with changes in `currentPaymentMethodInterface` not surfacing to payment methods. This could explain some of the unusual behaviour reported in testing.  The reason being that if there are any changes in the data exposed through the `paymentMethodInterface` that also doesn't impact any of the dependencies for this effect, then the payment method element will be still using stale data from the interface.